client/components/data/query-source-payment-transaction-detail/index.jsx

```diff
@@ -0,0 +1,60 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { fetchSourcePaymentTransactionDetail } from 'state/transactions/source-payment/actions';
+import { getSourcePaymentTransactionDetail } from 'state/selectors';
+
+class QuerySourcePaymentTransactionDetail extends React.Component {
+	static propTypes = {
+		pollIntervalMs: PropTypes.number,
```

Comment:
Should `fetchTransactionDetail` be a (required) prop? I know it's provided by `mapDispatchToProps` below, but the component doesn't know that. 😁 (And it will throw an exception if it does not exist when it mounts.)

What about `orderId` and `transactionDetail`? Those seem to be equally important.